usaspending_api/etl/subaward_etl.py

```diff
@@ -60,10 +64,18 @@ def load_subawards(submission_attributes, db_cursor):
                 d1_empty_count += 1
             continue
 
+        # Get the agency
+        agency = get_valid_awarding_agency(row)
+
+        if not agency:
+            logger.warn("Subaward number {} cannot find matching agency with toptier code {} and subtier code {}".format(row['subcontract_num'], row['awarding_agency_code'], row['awarding_sub_tier_agency_c']))
+            continue
+
         # Find the award to attach this sub-contract to
         # We perform this lookup by finding the Award containing a transaction with
         # a matching parent award id, piid, and submission attributes
-        award = Award.objects.filter(transaction__submission=submission_attributes,
+        award = Award.objects.filter(awarding_agency=agency,
+                                     transaction__submission=submission_attributes,
                                      transaction__contract_data__piid=row['piid'],
                                      transaction__contract_data__isnull=False,
                                      transaction__contract_data__parent_award_id=row['parent_award_id']).distinct().first()
```

Comment:
Apologies for not noticing this before, but what do you think about adding a `sort_by` here as a hedge against some kind of data malfunction?